version: '3'
services:
  postgres:
    image: postgres:9.6
    container_name: gen3-fence-postgres
    networks:
      - devnet
    volumes:
      - psqldata:/var/lib/postgresql/data
    restart: unless-stopped
    environment:
      - POSTGRES_PASSWORD=postgres
    ports:
      - 5432:5432
  pgadmin:
    container_name: gen3-fence-pgadmin
    image: dpage/pgadmin4
    restart: always
    networks:
      - devnet
    environment:
      - PGADMIN_DEFAULT_EMAIL=admin@wisecode.ai
      - PGADMIN_DEFAULT_PASSWORD=Welcometo2021!
    ports:
      - 5050:80
    volumes:
      - pgadmindata:/var/lib/pgadmin
  fence-service:
    build: .
    container_name: fence-service
    networks:
      - devnet
    volumes:
      - ./Secrets/TLS/service.crt:/usr/local/share/ca-certificates/cdis-ca.crt
      - ./keys/dev:/fence/keys/dev
      - ./fence-config.yaml:/fence/fence-config.yaml
    ports:
      - 443:443
      - 80:80
    healthcheck:
      test: [ "CMD-SHELL", "curl -f http://localhost/_status" ]
      interval: 60s
      timeout: 5s
      retries: 3
    environment:
      - PYTHONPATH=/var/www/fence
      - USER_API=http://localhost/user
      - GEN3_DEBUG=True
      - USER_SERVICE_BASE_URL=https://cngooq0hll.execute-api.us-east-2.amazonaws.com/user/
      - FENCE_BASE_URL=http://localhost:8000/
    depends_on:
      - postgres

networks:
  devnet:
volumes:
  psqldata:
  pgadmindata: